common/persistence/pinotVisibilityTripleManager.go

```diff
@@ -336,13 +339,24 @@ func (v *pinotVisibilityTripleManager) logUserQueryParameters(userParam userPara
 		return
 	}
 
-	v.logger.Info("Logging user query parameters for Pinot/ES response comparator...",
-		tag.OperationName(userParam.operation),
-		tag.WorkflowDomainName(userParam.domainName),
-		tag.WorkflowType(userParam.workflowType),
-		tag.WorkflowID(userParam.workflowID),
-		tag.WorkflowCloseStatus(userParam.closeStatus),
-		tag.VisibilityQuery(filterAttrPrefix(userParam.customQuery)))
+	if userParam.earlistTime == nil && userParam.latestTime == nil {
+		v.logger.Info("Logging user query parameters for Pinot/ES response comparator with custom query...",
```

Comment:
I'm nitpicking, but I'd suggest pulling the responsibility for handling nil behaviour into the tag (as per the tag discussion) and avoiding branching an if here